Index: samba-3.0.21a.new/source/include/includes.h
===================================================================
--- samba-3.0.21a.new.orig/source/include/includes.h	2006-01-02 09:35:13.800068024 +0100
+++ samba-3.0.21a.new/source/include/includes.h	2006-01-02 09:35:22.030398495 +0100
@@ -1049,10 +1049,7 @@
 #endif
 
 #ifndef DEFAULT_PRINTING
-#ifdef HAVE_CUPS
-#define DEFAULT_PRINTING PRINT_CUPS
-#define PRINTCAP_NAME "cups"
-#elif defined(SYSV)
+#if defined(SYSV)
 #define DEFAULT_PRINTING PRINT_SYSV
 #define PRINTCAP_NAME "lpstat"
 #else
