Index: samba-3.0.21c/source/include/includes.h
===================================================================
--- samba-3.0.21c.orig/source/include/includes.h	2006-02-25 11:57:11.836310142 +0100
+++ samba-3.0.21c/source/include/includes.h	2006-02-25 11:58:13.919783766 +0100
@@ -1068,10 +1068,7 @@
 #endif
 
 #ifndef DEFAULT_PRINTING
-#ifdef HAVE_CUPS
-#define DEFAULT_PRINTING PRINT_CUPS
-#define PRINTCAP_NAME "cups"
-#elif defined(SYSV)
+#if defined(SYSV)
 #define DEFAULT_PRINTING PRINT_SYSV
 #define PRINTCAP_NAME "lpstat"
 #else
