diff -ur samba-2.99.cvs.20020713.orig/source/acconfig.h samba-2.99.cvs.20020713/source/acconfig.h
--- samba-2.99.cvs.20020713.orig/source/acconfig.h	Fri May 17 07:42:29 2002
+++ samba-2.99.cvs.20020713/source/acconfig.h	Sat Jul 13 13:41:36 2002
@@ -188,6 +188,8 @@
 #undef STAT_ST_BLOCKSIZE
 #undef HAVE_DEVICE_MAJOR_FN
 #undef HAVE_DEVICE_MINOR_FN
+#undef FHS_COMPATIBLE
+#undef VARDIR
 #undef HAVE_PASSWD_PW_COMMENT
 #undef HAVE_PASSWD_PW_AGE
 /*
diff -ur samba-2.99.cvs.20020713.orig/source/configure.in samba-2.99.cvs.20020713/source/configure.in
--- samba-2.99.cvs.20020713.orig/source/configure.in	Sat Jul 13 13:36:38 2002
+++ samba-2.99.cvs.20020713/source/configure.in	Sat Jul 13 13:43:11 2002
@@ -12,11 +12,13 @@
 AC_ARG_WITH(fhs, 
 [  --with-fhs              Use FHS-compliant paths (default=no)],
     configdir="${sysconfdir}/samba"
-    lockdir="\${VARDIR}/cache/samba"
+    lockdir="\${VARDIR}/run/samba"
     piddir="\$(VARDIR)/run/samba"
     logfilebase="\${VARDIR}/log/samba"
     privatedir="\${CONFIGDIR}/private"
-    swatdir="\${DATADIR}/samba/swat",
+    swatdir="\${DATADIR}/samba/swat"
+    AC_DEFINE(FHS_COMPATIBLE)
+    AC_DEFINE_UNQUOTED(VARDIR,"$localstatedir"),
     configdir="\$(LIBDIR)"
     logfilebase="\$(VARDIR)"
     lockdir="\${VARDIR}/locks"
diff -ur samba-2.99.cvs.20020713.orig/source/groupdb/mapping.c samba-2.99.cvs.20020713/source/groupdb/mapping.c
--- samba-2.99.cvs.20020713.orig/source/groupdb/mapping.c	Fri Jun  7 09:33:30 2002
+++ samba-2.99.cvs.20020713/source/groupdb/mapping.c	Sat Jul 13 13:56:02 2002
@@ -214,7 +214,7 @@
 	
 	if (tdb && local_pid == sys_getpid())
 		return True;
-	tdb = tdb_open_log(lock_path("group_mapping.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	tdb = tdb_open_log(state_path("group_mapping.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!tdb) {
 		DEBUG(0,("Failed to open group mapping database\n"));
 		return False;
diff -ur samba-2.99.cvs.20020713.orig/source/include/config.h.in samba-2.99.cvs.20020713/source/include/config.h.in
--- samba-2.99.cvs.20020713.orig/source/include/config.h.in	Fri May 17 07:42:34 2002
+++ samba-2.99.cvs.20020713/source/include/config.h.in	Sat Jul 13 13:44:09 2002
@@ -254,6 +254,8 @@
 #undef STAT_ST_BLOCKSIZE
 #undef HAVE_DEVICE_MAJOR_FN
 #undef HAVE_DEVICE_MINOR_FN
+#undef FHS_COMPATIBLE
+#undef VARDIR
 #undef HAVE_PASSWD_PW_COMMENT
 #undef HAVE_PASSWD_PW_AGE
 /*
diff -ur samba-2.99.cvs.20020713.orig/source/include/local.h samba-2.99.cvs.20020713/source/include/local.h
--- samba-2.99.cvs.20020713.orig/source/include/local.h	Fri Apr 12 03:22:50 2002
+++ samba-2.99.cvs.20020713/source/include/local.h	Sat Jul 13 13:39:15 2002
@@ -203,4 +203,20 @@
 /* this enables the "rabbit pellet" fix for SMBwritebraw */
 #define RABBIT_PELLET_FIX 1
 
+/* FHS-compatible directory defines */
+#ifdef FHS_COMPATIBLE
+#ifndef CACHEDIR
+#define CACHEDIR VARDIR "/cache/samba"
+#endif
+#ifndef STATEDIR
+#define STATEDIR VARDIR "/lib/samba"
+#endif
+
+#else
+
+#define CACHEDIR lp_lockdir()
+#define STATEDIR lp_lockdir()
+
+#endif
+
 #endif
diff -ur samba-2.99.cvs.20020713.orig/source/intl/lang_tdb.c samba-2.99.cvs.20020713/source/intl/lang_tdb.c
--- samba-2.99.cvs.20020713.orig/source/intl/lang_tdb.c	Wed Jan 30 00:08:16 2002
+++ samba-2.99.cvs.20020713/source/intl/lang_tdb.c	Sat Jul 13 14:01:20 2002
@@ -129,7 +129,7 @@
 	}
 	
 
-	asprintf(&path, "%s%s.tdb", lock_path("lang_"), lang);
+	asprintf(&path, "%s%s.tdb", state_path("lang_"), lang);
 
 	tdb = tdb_open_log(path, 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0644);
 	if (!tdb) {
diff -ur samba-2.99.cvs.20020713.orig/source/lib/account_pol.c samba-2.99.cvs.20020713/source/lib/account_pol.c
--- samba-2.99.cvs.20020713.orig/source/lib/account_pol.c	Wed Jan 30 00:08:16 2002
+++ samba-2.99.cvs.20020713/source/lib/account_pol.c	Sat Jul 13 14:01:35 2002
@@ -34,7 +34,7 @@
 
 	if (tdb && local_pid == sys_getpid())
 		return True;
-	tdb = tdb_open_log(lock_path("account_policy.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	tdb = tdb_open_log(state_path("account_policy.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!tdb) {
 		DEBUG(0,("Failed to open account policy database\n"));
 		return False;
diff -ur samba-2.99.cvs.20020713.orig/source/lib/util.c samba-2.99.cvs.20020713/source/lib/util.c
--- samba-2.99.cvs.20020713.orig/source/lib/util.c	Wed Jul 10 19:06:26 2002
+++ samba-2.99.cvs.20020713/source/lib/util.c	Sat Jul 13 13:39:15 2002
@@ -1911,6 +1911,46 @@
 	return fname;
 }
 
+/*****************************************************************
+a useful function for returning a path in the Samba state directory
+ *****************************************************************/
+char *state_path(char *name)
+{
+	static pstring fname;
+
+	pstrcpy(fname,STATEDIR);
+	trim_string(fname,"","/");
+
+	if (!directory_exist(fname,NULL)) {
+		mkdir(fname,0755);
+	}
+
+	pstrcat(fname,"/");
+	pstrcat(fname,name);
+
+	return fname;
+}
+
+/*****************************************************************
+a useful function for returning a path in the Samba cache directory
+ *****************************************************************/
+char *cache_path(char *name)
+{
+	static pstring fname;
+
+	pstrcpy(fname,CACHEDIR);
+	trim_string(fname,"","/");
+
+	if (!directory_exist(fname,NULL)) {
+		mkdir(fname,0755);
+	}
+
+	pstrcat(fname,"/");
+	pstrcat(fname,name);
+
+	return fname;
+}
+
 /*******************************************************************
  Given a filename - get its directory name
  NB: Returned in static storage.  Caveats:
diff -ur samba-2.99.cvs.20020713.orig/source/libsmb/netlogon_unigrp.c samba-2.99.cvs.20020713/source/libsmb/netlogon_unigrp.c
--- samba-2.99.cvs.20020713.orig/source/libsmb/netlogon_unigrp.c	Fri Mar 15 03:19:51 2002
+++ samba-2.99.cvs.20020713/source/libsmb/netlogon_unigrp.c	Sat Jul 13 14:03:07 2002
@@ -40,7 +40,7 @@
 BOOL uni_group_cache_init(void)
 {
 	if (!netlogon_unigrp_tdb) {
-		netlogon_unigrp_tdb = tdb_open_log(lock_path("netlogon_unigrp.tdb"), 0,
+		netlogon_unigrp_tdb = tdb_open_log(state_path("netlogon_unigrp.tdb"), 0,
 						   TDB_DEFAULT, O_RDWR | O_CREAT, 0644);
 	}
 
@@ -110,7 +110,7 @@
 		return NULL;
 	}
 	if (!netlogon_unigrp_tdb) {
-		netlogon_unigrp_tdb = tdb_open_log(lock_path("netlogon_unigrp.tdb"), 0,
+		netlogon_unigrp_tdb = tdb_open_log(state_path("netlogon_unigrp.tdb"), 0,
                 				    TDB_DEFAULT, O_RDWR, 0644);
 	}
 	if (!netlogon_unigrp_tdb) {
diff -ur samba-2.99.cvs.20020713.orig/source/nmbd/nmbd_serverlistdb.c samba-2.99.cvs.20020713/source/nmbd/nmbd_serverlistdb.c
--- samba-2.99.cvs.20020713.orig/source/nmbd/nmbd_serverlistdb.c	Wed Jan 30 00:08:22 2002
+++ samba-2.99.cvs.20020713/source/nmbd/nmbd_serverlistdb.c	Sat Jul 13 13:39:15 2002
@@ -347,7 +347,7 @@
 
   updatecount++;
     
-  pstrcpy(fname,lp_lockdir());
+  pstrcpy(fname,CACHEDIR);
   trim_string(fname,NULL,"/");
   pstrcat(fname,"/");
   pstrcat(fname,SERVER_LIST);
diff -ur samba-2.99.cvs.20020713.orig/source/nmbd/nmbd_winsserver.c samba-2.99.cvs.20020713/source/nmbd/nmbd_winsserver.c
--- samba-2.99.cvs.20020713.orig/source/nmbd/nmbd_winsserver.c	Mon Jul  1 02:19:12 2002
+++ samba-2.99.cvs.20020713/source/nmbd/nmbd_winsserver.c	Sat Jul 13 13:45:43 2002
@@ -234,7 +234,7 @@
 
 	add_samba_names_to_subnet(wins_server_subnet);
 
-	tdb = tdb_open_log(lock_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
+	tdb = tdb_open_log(state_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
 	if (!tdb) {
 		DEBUG(2,("initialise_wins: Can't open wins database file %s. Error was %s\n", WINS_LIST, strerror(errno) ));
 		return True;
diff -ur samba-2.99.cvs.20020713.orig/source/nsswitch/winbindd_ads.c samba-2.99.cvs.20020713/source/nsswitch/winbindd_ads.c
--- samba-2.99.cvs.20020713.orig/source/nsswitch/winbindd_ads.c	Thu Jul 11 00:28:08 2002
+++ samba-2.99.cvs.20020713/source/nsswitch/winbindd_ads.c	Sat Jul 13 14:04:53 2002
@@ -116,7 +116,7 @@
 	}
 
 	/* we don't want this to affect the users ccache */
-	ccache = lock_path("winbindd_ccache");
+	ccache = cache_path("winbindd_ccache");
 	SETENV("KRB5CCNAME", ccache, 1);
 	unlink(ccache);
 
diff -ur samba-2.99.cvs.20020713.orig/source/nsswitch/winbindd_cache.c samba-2.99.cvs.20020713/source/nsswitch/winbindd_cache.c
--- samba-2.99.cvs.20020713.orig/source/nsswitch/winbindd_cache.c	Tue Jun 18 04:20:08 2002
+++ samba-2.99.cvs.20020713/source/nsswitch/winbindd_cache.c	Sat Jul 13 13:46:25 2002
@@ -53,7 +53,7 @@
 	}
 	if (opt_nocache) return;
 
-	wcache->tdb = tdb_open_log(lock_path("winbindd_cache.tdb"), 5000, 
+	wcache->tdb = tdb_open_log(cache_path("winbindd_cache.tdb"), 5000, 
 				   TDB_CLEAR_IF_FIRST, O_RDWR|O_CREAT, 0600);
 
 	if (!wcache->tdb) {
diff -ur samba-2.99.cvs.20020713.orig/source/nsswitch/winbindd_idmap.c samba-2.99.cvs.20020713/source/nsswitch/winbindd_idmap.c
--- samba-2.99.cvs.20020713.orig/source/nsswitch/winbindd_idmap.c	Tue Jun 18 04:20:09 2002
+++ samba-2.99.cvs.20020713/source/nsswitch/winbindd_idmap.c	Sat Jul 13 13:39:15 2002
@@ -430,14 +430,14 @@
 {
 	/* Open tdb cache */
 
-	if (!(idmap_tdb = tdb_open_log(lock_path("winbindd_idmap.tdb"), 0,
+	if (!(idmap_tdb = tdb_open_log(state_path("winbindd_idmap.tdb"), 0,
 				TDB_DEFAULT, O_RDWR | O_CREAT, 0600))) {
 		DEBUG(0, ("winbindd_idmap_init: Unable to open idmap database\n"));
 		return False;
 	}
 
 	/* possibly convert from an earlier version */
-	if (!idmap_convert(lock_path("winbindd_idmap.tdb"))) {
+	if (!idmap_convert(state_path("winbindd_idmap.tdb"))) {
 		DEBUG(0, ("winbindd_idmap_init: Unable to open idmap database\n"));
 		return False;
 	}
diff -ur samba-2.99.cvs.20020713.orig/source/param/loadparm.c samba-2.99.cvs.20020713/source/param/loadparm.c
--- samba-2.99.cvs.20020713.orig/source/param/loadparm.c	Wed Jul 10 19:06:27 2002
+++ samba-2.99.cvs.20020713/source/param/loadparm.c	Sat Jul 13 13:47:30 2002
@@ -95,6 +95,9 @@
 	char *szAddPrinterCommand;
 	char *szDeletePrinterCommand;
 	char *szOs2DriverMap;
+#ifdef FHS_COMPATIBLE
+	char *szLockDirStub;
+#endif
 	char *szLockDir;
 	char *szPidDir;
 	char *szRootdir;
@@ -966,8 +969,13 @@
 	{"config file", P_STRING, P_GLOBAL, &Globals.szConfigFile, NULL, NULL, FLAG_HIDE},
 	{"preload", P_STRING, P_GLOBAL, &Globals.szAutoServices, NULL, NULL, 0},
 	{"auto services", P_STRING, P_GLOBAL, &Globals.szAutoServices, NULL, NULL, 0},
+#ifdef FHS_COMPATIBLE
+	{"lock dir", P_STRING, P_GLOBAL, &Globals.szLockDirStub, NULL, NULL, 0},
+	{"lock directory", P_STRING, P_GLOBAL, &Globals.szLockDirStub, NULL, NULL, 0},
+#else
 	{"lock dir", P_STRING, P_GLOBAL, &Globals.szLockDir, NULL, NULL, 0}, 
 	{"lock directory", P_STRING, P_GLOBAL, &Globals.szLockDir, NULL, NULL, 0},
+#endif
 	{"pid directory", P_STRING, P_GLOBAL, &Globals.szPidDir, NULL, NULL, 0}, 
 #ifdef WITH_UTMP
 	{"utmp directory", P_STRING, P_GLOBAL, &Globals.szUtmpDir, NULL, NULL, 0},
diff -ur samba-2.99.cvs.20020713.orig/source/passdb/secrets.c samba-2.99.cvs.20020713/source/passdb/secrets.c
--- samba-2.99.cvs.20020713.orig/source/passdb/secrets.c	Thu May 23 10:42:29 2002
+++ samba-2.99.cvs.20020713/source/passdb/secrets.c	Sat Jul 13 13:50:55 2002
@@ -37,8 +37,7 @@
 	if (tdb)
 		return True;
 
-	pstrcpy(fname, lp_private_dir());
-	pstrcat(fname,"/secrets.tdb");
+	pstrcpy(fname,state_path("secrets.tdb"));
 
 	tdb = tdb_open_log(fname, 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 
diff -ur samba-2.99.cvs.20020713.orig/source/printing/nt_printing.c samba-2.99.cvs.20020713/source/printing/nt_printing.c
--- samba-2.99.cvs.20020713.orig/source/printing/nt_printing.c	Thu Jul 11 08:17:56 2002
+++ samba-2.99.cvs.20020713/source/printing/nt_printing.c	Sat Jul 13 13:39:15 2002
@@ -263,24 +263,24 @@
 	if (tdb_drivers && tdb_printers && tdb_forms && local_pid == sys_getpid())
 		return True;
  
-	tdb_drivers = tdb_open_log(lock_path("ntdrivers.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	tdb_drivers = tdb_open_log(state_path("ntdrivers.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!tdb_drivers) {
 		DEBUG(0,("nt_printing_init: Failed to open nt drivers database %s (%s)\n",
-			lock_path("ntdrivers.tdb"), strerror(errno) ));
+			state_path("ntdrivers.tdb"), strerror(errno) ));
 		return False;
 	}
  
-	tdb_printers = tdb_open_log(lock_path("ntprinters.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	tdb_printers = tdb_open_log(state_path("ntprinters.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!tdb_printers) {
 		DEBUG(0,("nt_printing_init: Failed to open nt printers database %s (%s)\n",
-			lock_path("ntprinters.tdb"), strerror(errno) ));
+			state_path("ntprinters.tdb"), strerror(errno) ));
 		return False;
 	}
  
-	tdb_forms = tdb_open_log(lock_path("ntforms.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	tdb_forms = tdb_open_log(state_path("ntforms.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!tdb_forms) {
 		DEBUG(0,("nt_printing_init: Failed to open nt forms database %s (%s)\n",
-			lock_path("ntforms.tdb"), strerror(errno) ));
+			state_path("ntforms.tdb"), strerror(errno) ));
 		return False;
 	}
  
diff -ur samba-2.99.cvs.20020713.orig/source/printing/printing.c samba-2.99.cvs.20020713/source/printing/printing.c
--- samba-2.99.cvs.20020713.orig/source/printing/printing.c	Wed Jul 10 19:06:29 2002
+++ samba-2.99.cvs.20020713/source/printing/printing.c	Sat Jul 13 13:48:56 2002
@@ -186,7 +186,7 @@
 		DLIST_ADD(print_db_head, p);
 	}
 
-	pstrcpy(printdb_path, lock_path(printername));
+	pstrcpy(printdb_path, cache_path(printername));
 	pstrcat(printdb_path, ".tdb");
 	p->tdb = tdb_open_log(printdb_path, 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!p->tdb) {
@@ -221,7 +221,7 @@
 	if (local_pid == sys_getpid())
 		return True;
 
-	unlink(lock_path("printing.tdb"));
+	unlink(cache_path("printing.tdb"));
 	local_pid = sys_getpid();
 
 	/* handle a Samba upgrade */
diff -ur samba-2.99.cvs.20020713.orig/source/rpc_server/srv_reg_nt.c samba-2.99.cvs.20020713/source/rpc_server/srv_reg_nt.c
--- samba-2.99.cvs.20020713.orig/source/rpc_server/srv_reg_nt.c	Sun Jul  7 20:06:38 2002
+++ samba-2.99.cvs.20020713/source/rpc_server/srv_reg_nt.c	Sat Jul 13 14:07:35 2002
@@ -325,13 +325,13 @@
 	 * if we need to init the data in the registry
 	 */
 	
-	tdb_reg = tdb_open_log(lock_path("registry.tdb"), 0, TDB_DEFAULT, O_RDWR, 0600);
+	tdb_reg = tdb_open_log(state_path("registry.tdb"), 0, TDB_DEFAULT, O_RDWR, 0600);
 	if ( !tdb_reg ) 
 	{
-		tdb_reg = tdb_open_log(lock_path("registry.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+		tdb_reg = tdb_open_log(state_path("registry.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 		if ( !tdb_reg ) {
 			DEBUG(0,("init_registry: Failed to open registry %s (%s)\n",
-				lock_path("registry.tdb"), strerror(errno) ));
+				state_path("registry.tdb"), strerror(errno) ));
 			return False;
 		}
 		
diff -ur samba-2.99.cvs.20020713.orig/source/rpc_server/srv_srvsvc_nt.c samba-2.99.cvs.20020713/source/rpc_server/srv_srvsvc_nt.c
--- samba-2.99.cvs.20020713.orig/source/rpc_server/srv_srvsvc_nt.c	Tue Jul  2 01:34:27 2002
+++ samba-2.99.cvs.20020713/source/rpc_server/srv_srvsvc_nt.c	Sat Jul 13 13:39:15 2002
@@ -127,10 +127,10 @@
  
 	if (share_tdb && local_pid == sys_getpid())
 		return True;
-	share_tdb = tdb_open_log(lock_path("share_info.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
+	share_tdb = tdb_open_log(state_path("share_info.tdb"), 0, TDB_DEFAULT, O_RDWR|O_CREAT, 0600);
 	if (!share_tdb) {
 		DEBUG(0,("Failed to open share info database %s (%s)\n",
-			lock_path("share_info.tdb"), strerror(errno) ));
+			state_path("share_info.tdb"), strerror(errno) ));
 		return False;
 	}
  
diff -ur samba-2.99.cvs.20020713.orig/source/smbd/lanman.c samba-2.99.cvs.20020713/source/smbd/lanman.c
--- samba-2.99.cvs.20020713.orig/source/smbd/lanman.c	Tue Jul  2 01:34:27 2002
+++ samba-2.99.cvs.20020713/source/smbd/lanman.c	Sat Jul 13 13:49:45 2002
@@ -1095,9 +1095,9 @@
   BOOL local_list_only;
   int i;
 
-  lines = file_lines_load(lock_path(SERVER_LIST), NULL);
+  lines = file_lines_load(cache_path(SERVER_LIST), NULL);
   if (!lines) {
-    DEBUG(4,("Can't open %s - %s\n",lock_path(SERVER_LIST),strerror(errno)));
+    DEBUG(4,("Can't open %s - %s\n",cache_path(SERVER_LIST),strerror(errno)));
     return(0);
   }
 
diff -ur samba-2.99.cvs.20020713.orig/source/wrepld/process.c samba-2.99.cvs.20020713/source/wrepld/process.c
--- samba-2.99.cvs.20020713.orig/source/wrepld/process.c	Fri Jun 14 11:02:59 2002
+++ samba-2.99.cvs.20020713/source/wrepld/process.c	Sat Jul 13 14:09:56 2002
@@ -197,7 +197,7 @@
 {
 	TDB_CONTEXT *tdb;
 
-	tdb = tdb_open_log(lock_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
+	tdb = tdb_open_log(state_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
 	if (!tdb) {
 		DEBUG(2,("get_our_last_id: Can't open wins database file %s. Error was %s\n", WINS_LIST, strerror(errno) ));
 		return;
@@ -489,7 +489,7 @@
 		}
 
 
-	tdb = tdb_open_log(lock_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
+	tdb = tdb_open_log(state_path(WINS_LIST), 0, TDB_DEFAULT, O_RDONLY, 0600);
 	if (!tdb) {
 		DEBUG(2,("send_entry_request: Can't open wins database file %s. Error was %s\n", WINS_LIST, strerror(errno) ));
 		return;
