Index: samba-3.0.21c/source/script/installswat.sh
===================================================================
--- samba-3.0.21c.orig/source/script/installswat.sh	2006-02-25 11:57:11.178442873 +0100
+++ samba-3.0.21c/source/script/installswat.sh	2006-02-25 11:58:18.019956453 +0100
@@ -114,7 +114,11 @@
 
 # Install Using Samba book (but only if it is there)
 
-if [ "x$BOOKDIR" != "x" -a -f $SRCDIR../docs/htmldocs/using_samba/toc.html ]; then
+# Under Debian we don't actually install the book. The book is part of
+# the samba-doc package, so we just provide a symlink that points to
+# where the book is actually installed. The symlink is created in
+# debian/rules.
+if /bin/false; then
 
     # Create directories
 
