Index: samba-3.0.21c/source/Makefile.in
===================================================================
--- samba-3.0.21c.orig/source/Makefile.in	2006-02-25 11:58:15.760412374 +0100
+++ samba-3.0.21c/source/Makefile.in	2006-02-25 11:58:18.819795065 +0100
@@ -1491,6 +1491,7 @@
 
 python_clean:
 	@-if test -n "$(PYTHON)"; then $(PYTHON) python/setup.py clean; fi
+	@-rm -rf build/
 
 # revert to the previously installed version
 revert:
@@ -1559,6 +1560,7 @@
 	@rm -f include/proto.h include/build_env.h  include/wrepld_proto.h \
 		nsswitch/winbindd_proto.h web/swat_proto.h \
 		client/client_proto.h utils/net_proto.h \
+	        utils/ntlm_auth_proto.h \
 		smbd/build_options.c
 
 MKPROTO_SH = $(srcdir)/script/mkproto.sh
@@ -1641,6 +1643,7 @@
 
 distclean: realclean
 	-rm -f include/stamp-h
+	-rm -f script/gen-8bit-gap.sh smbadduser
 	-rm -f include/config.h Makefile
 	-rm -f config.status config.cache so_locations
 	-rm -rf .deps TAGS
