diff -ur samba-2.99.cvs.20020713.orig/source/Makefile.in samba-2.99.cvs.20020713/source/Makefile.in
--- samba-2.99.cvs.20020713.orig/source/Makefile.in	Sat Jul 13 04:14:39 2002
+++ samba-2.99.cvs.20020713/source/Makefile.in	Sat Jul 13 13:28:35 2002
@@ -831,7 +831,8 @@
 	@$(SHELL) $(srcdir)/script/installswat.sh $(SWATDIR) $(srcdir)
 
 installclientlib:
-	-$(INSTALLCMD) bin/libsmbclient.so ${prefix}/lib
+	-$(INSTALLCMD) bin/libsmbclient.so $(BASEDIR)/lib
+	-$(INSTALLCMD) bin/libsmbclient.a $(BASEDIR)/lib
 	-$(INSTALLCMD) -d ${prefix}/include
 	-$(INSTALLCMD) include/libsmbclient.h ${prefix}/include
 
