Index: samba-3.0.23/source/Makefile.in
===================================================================
--- samba-3.0.23.orig/source/Makefile.in	2006-07-11 08:59:29.326658704 +0200
+++ samba-3.0.23/source/Makefile.in	2006-07-11 08:59:32.806682604 +0200
@@ -1521,6 +1521,7 @@
 
 python_clean:
 	@-if test -n "$(PYTHON)"; then $(PYTHON) python/setup.py clean; fi
+	@-rm -rf build/
 
 # revert to the previously installed version
 revert:
@@ -1698,6 +1699,7 @@
 
 distclean: realclean
 	-rm -f include/stamp-h
+	-rm -f script/gen-8bit-gap.sh smbadduser
 	-rm -f include/config.h Makefile
 	-rm -f config.status config.cache so_locations
 	-rm -rf .deps TAGS
