Index: samba-3.0.21a.new/source/Makefile.in
===================================================================
--- samba-3.0.21a.new.orig/source/Makefile.in	2006-01-02 09:43:36.349338911 +0100
+++ samba-3.0.21a.new/source/Makefile.in	2006-01-02 09:58:46.622122510 +0100
@@ -1490,6 +1490,7 @@
 
 python_clean:
 	@-if test -n "$(PYTHON)"; then $(PYTHON) python/setup.py clean; fi
+	@-rm -rf build/
 
 # revert to the previously installed version
 revert:
@@ -1558,6 +1559,7 @@
 	@rm -f include/proto.h include/build_env.h  include/wrepld_proto.h \
 		nsswitch/winbindd_proto.h web/swat_proto.h \
 		client/client_proto.h utils/net_proto.h \
+	        utils/ntlm_auth_proto.h \
 		smbd/build_options.c
 
 MKPROTO_SH = $(srcdir)/script/mkproto.sh
@@ -1634,6 +1636,7 @@
 
 distclean: realclean
 	-rm -f include/stamp-h
+	-rm -f script/gen-8bit-gap.sh smbadduser
 	-rm -f include/config.h Makefile
 	-rm -f config.status config.cache so_locations
 	-rm -rf .deps TAGS
