--- samba-2.2.8/source/Makefile.in.orig	2003-02-28 10:56:06.000000000 -0500
+++ samba-2.2.8/source/Makefile.in	2003-03-16 15:20:08.000000000 -0500
@@ -727,8 +727,8 @@
 	@$(SHELL) $(srcdir)/script/installswat.sh $(SWATDIR) $(srcdir)
 
 installclientlib:
-	-$(INSTALLCLIENTCMD_SH) bin/libsmbclient.@SHLIBEXT@
-	-$(INSTALLCLIENTCMD_A) bin/libsmbclient.a
+	-$(INSTALLCMD) bin/libsmbclient.@SHLIBEXT@ ${prefix}/lib
+	-$(INSTALLCMD) bin/libsmbclient.a ${prefix}/lib
 	-$(INSTALLCMD) -d ${prefix}/include
 	-$(INSTALLCMD) include/libsmbclient.h ${prefix}/include
 
