diff -ur samba-2.99.cvs.20020713.orig/source/script/installman.sh samba-2.99.cvs.20020713/source/script/installman.sh
--- samba-2.99.cvs.20020713.orig/source/script/installman.sh	Thu Jul 18 22:56:55 2002
+++ samba-2.99.cvs.20020713/source/script/installman.sh	Thu Jul 18 22:58:46 2002
@@ -22,8 +22,8 @@
 	echo Installing \"$lang\" man pages in $MANDIR/lang/$lang
     fi
 
-    langdir=$MANDIR/lang/$lang
-    for d in $MANDIR $MANDIR/lang $langdir $langdir/man1 $langdir/man5 $langdir/man7 $langdir/man8; do
+    langdir=$MANDIR/$lang
+    for d in $MANDIR $langdir $langdir/man1 $langdir/man5 $langdir/man7 $langdir/man8; do
 	if [ ! -d $d ]; then
 	    mkdir $d
 	    if [ ! -d $d ]; then
