Index: samba-3.0.21c/source/pam_smbpass/README
===================================================================
--- samba-3.0.21c.orig/source/pam_smbpass/README	2006-02-25 11:57:10.632553011 +0100
+++ samba-3.0.21c/source/pam_smbpass/README	2006-02-25 11:58:22.133126513 +0100
@@ -37,7 +37,7 @@
 	smbconf=<file>	-	specify an alternate path to the smb.conf
 				file.
 
-See the samples/ directory for example PAM configurations using this
+See the examples/ directory for example PAM configurations using this
 module.
 
 Thanks go to the following people:
