Index: samba-3.0.23pre1/source/python/py_tdb.c
===================================================================
--- samba-3.0.23pre1.orig/source/python/py_tdb.c	2006-05-24 20:31:14.368997592 +0200
+++ samba-3.0.23pre1/source/python/py_tdb.c	2006-05-24 20:32:02.528676208 +0200
@@ -494,7 +494,7 @@
 	if (!PyArg_ParseTuple(args, "s|i", &s, &timeout))
 		return NULL;
 
-	/*	result = tdb_lock_bystring(obj->tdb, s, timeout); */
+		result = tdb_lock_bystring_with_timeout(obj->tdb, s, timeout); 
 
 	return PyInt_FromLong(result != -1);
 }
