Index: samba-3.0.21a.new/examples/LDAP/README
===================================================================
--- samba-3.0.21a.new.orig/examples/LDAP/README	2005-02-25 18:59:51.000000000 +0100
+++ samba-3.0.21a.new/examples/LDAP/README	2006-01-02 09:53:32.809543869 +0100
@@ -59,6 +59,9 @@
 The smbldap-tools package can be downloaded individually from 
 http://samba.idealx.org/dist/
 
+On Debian systems, the smbldap-tools exists as a separate package
+and is not included in LDAP examples.
+
 !==
 !== end of README
 !==
