--- samba/docs/docbook/manpages/smb.conf.5.sgml.orig	Thu May  2 13:49:14 2002
+++ samba/docs/docbook/manpages/smb.conf.5.sgml	Thu May  2 14:03:45 2002
@@ -4711,7 +4711,7 @@
 
 		<para>Here's a way of sending the messages as mail to root:</para>
 
-		<para><command>message command = /bin/mail -s 'message from %f on 
+		<para><command>message command = /usr/bin/mail -s 'message from %f on 
 		%m' root &lt; %s; rm %s</command></para>
 
 		<para>If you don't have a message command then the message 
