Index: samba-3.0.21c/source/include/local.h
===================================================================
--- samba-3.0.21c.orig/source/include/local.h	2006-02-25 11:57:10.314617157 +0100
+++ samba-3.0.21c/source/include/local.h	2006-02-25 11:58:24.566635487 +0100
@@ -109,7 +109,7 @@
 /* the default pager to use for the client "more" command. Users can
    override this with the PAGER environment variable */
 #ifndef PAGER
-#define PAGER "more"
+#define PAGER "/usr/bin/pager"
 #endif
 
 /* the size of the uid cache used to reduce valid user checks */
