Index: samba-3.0.23rc1/source/include/local.h
===================================================================
--- samba-3.0.23rc1.orig/source/include/local.h	2006-05-25 19:34:58.701380584 +0200
+++ samba-3.0.23rc1/source/include/local.h	2006-05-25 19:38:35.971350528 +0200
@@ -109,7 +109,7 @@
 /* the default pager to use for the client "more" command. Users can
    override this with the PAGER environment variable */
 #ifndef PAGER
-#define PAGER "more"
+#define PAGER "/usr/bin/pager"
 #endif
 
 /* the size of the uid cache used to reduce valid user checks */
