Index: samba-3.0.21c/source/client/smbmnt.c
===================================================================
--- samba-3.0.21c.orig/source/client/smbmnt.c	2006-02-25 11:57:10.224635312 +0100
+++ samba-3.0.21c/source/client/smbmnt.c	2006-02-25 11:58:25.369473493 +0100
@@ -276,7 +276,10 @@
         ment.mnt_fsname = share_name ? share_name : "none";
         ment.mnt_dir = mount_point;
         ment.mnt_type = "smbfs";
-        ment.mnt_opts = "";
+	if (mount_ro)
+                ment.mnt_opts = "ro";
+        else
+                ment.mnt_opts = "rw";
         ment.mnt_freq = 0;
         ment.mnt_passno= 0;
 
