Index: samba-3.0.21a.new/source/client/smbmount.c
===================================================================
--- samba-3.0.21a.new.orig/source/client/smbmount.c	2006-01-02 09:57:48.391891301 +0100
+++ samba-3.0.21a.new/source/client/smbmount.c	2006-01-02 09:57:49.591648836 +0100
@@ -211,6 +211,10 @@
   		c->capabilities &= ~CAP_STATUS32;
 		c->force_dos_errors = True;
 	}
+	/* For now, respect requests for uid-flattening mount options
+	   by disabling Unix permissions handling in the kernel driver */
+	if (mount_uid || mount_gid || mount_fmask || mount_dmask)
+		c->capabilities &= ~CAP_UNIX;
 
 	if (!cli_session_setup(c, username, 
 			       password, strlen(password),
