Index: samba-3.0.23rc3/source/utils/status.c
===================================================================
--- samba-3.0.23rc3.orig/source/utils/status.c	2006-06-24 13:46:15.856756500 +0200
+++ samba-3.0.23rc3/source/utils/status.c	2006-06-24 13:46:59.103459250 +0200
@@ -714,6 +714,16 @@
 	if ( show_locks ) {
 		int ret;
 
+		tdb = tdb_open_log(lock_path("locking.tdb"), 0, TDB_DEFAULT, O_RDONLY, 0);
+
+		if (!tdb) {
+			d_printf("%s not initialised\n", lock_path("locking.tdb"));
+			d_printf("This is normal if an SMB client has never connected to your server.\n");
+			exit(0);
+		} else {
+			tdb_close(tdb);
+		}
+
 		if (!locking_init(1)) {
 			d_printf("Can't initialise locking module - exiting\n");
 			exit(1);
