diff -ur samba-2.99.cvs.20020713.orig/source/client/clitar.c samba-2.99.cvs.20020713/source/client/clitar.c
--- samba-2.99.cvs.20020713.orig/source/client/clitar.c	Fri Jul 12 22:51:28 2002
+++ samba-2.99.cvs.20020713/source/client/clitar.c	Sat Jul 13 15:20:17 2002
@@ -492,7 +492,7 @@
   if (!*s2 && (*s1 == '/' || *s1 == '\\') && !*(s1+1)) return 0;
 
   /* check for s1 is an "initial" string of s2 */
-  if (*s2 == '/' || *s2 == '\\') return 0;
+  if ((*s2 == '/' || *s2 == '\\') && !*s1) return 0;
 
   return *s1-*s2;
 }
