Index: samba-3.0.23pre1/source/VERSION
===================================================================
--- samba-3.0.23pre1.orig/source/VERSION	2006-05-22 21:43:52.088487192 -0500
+++ samba-3.0.23pre1/source/VERSION	2006-05-22 21:45:34.968847000 -0500
@@ -96,5 +96,7 @@
 # e.g. SAMBA_VERSION_VENDOR_SUFFIX=vendor_version()    #
 #  ->  "CVS 3.0.0rc2-VendorVersion"                    #
 ########################################################
-SAMBA_VERSION_VENDOR_SUFFIX=
+SAMBA_VERSION_VENDOR_SUFFIX="Debian"
 SAMBA_VENDOR_PATCH=
+MOUNT_CIFS_VENDOR_SUFFIX="Debian"
+UNMOUNT_CIFS_VENDOR_SUFFIX="Debian"
