diff -ur samba-2.2.3a.orig/source/printing/lpq_parse.c samba-2.2.3a/source/printing/lpq_parse.c
--- samba-2.2.3a.orig/source/printing/lpq_parse.c	Sat Feb  2 18:46:49 2002
+++ samba-2.2.3a/source/printing/lpq_parse.c	Wed Mar 27 17:50:15 2002
@@ -266,6 +266,9 @@
     buf->status = LPQ_PAUSED;
   }
 
+  if(strequal(tokarr[LPRNG_RANKTOK],"done"))
+    buf->status = LPQ_PRINTED;
+
   buf->priority = *tokarr[LPRNG_PRIOTOK] -'A';
 
   buf->time = LPRng_time(tokarr[LPRNG_TIMETOK]);
